cmd/kube-apiserver/app/server.go

```diff
@@ -438,6 +440,14 @@ func buildGenericConfig(
 	lastErr error,
 ) {
 	genericConfig = genericapiserver.NewConfig(legacyscheme.Codecs)
+	genericConfig.IsTerminating = func() bool {
```

Comment:
i suppose it sort of breaks the "option->config->run" flow, pass the stop-channel when constructing config instance looks a bit odd..